client/src/cordova/apple/xcode/ios/Outline/Classes/AppKitBridge/AppKitController.swift

```diff
@@ -20,9 +20,30 @@ class AppKitController: NSObject {
 
     override public required init() {
         super.init()
-
-        // Indicates that the application is an ordinary app that appears in the Dock and may have a user interface.
         NSApp.setActivationPolicy(.regular)
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(windowWillClose(_:)),
+            name: NSWindow.willCloseNotification,
+            object: nil
+        )
+    }
+
+    @objc public func windowWillClose(_ notification: Notification) {
+        guard let closingWindow = notification.object as? NSWindow else {
+            return
+        }
+        // Are there any other visible "main" windows?
+        // (this notification also fires for menu popup and status windows)
+        let hasOtherVisibleWindows = NSApp.windows.contains { window in
+            return window !== closingWindow &&
+                   window.isVisible &&
+                   window.canBecomeMain &&
+                   !window.isMiniaturized
+        }
+        if !hasOtherVisibleWindows {
+            NSApp.setActivationPolicy(.accessory)
```

Comment:
If you are hiding on window close, you should show the dock item on window open. Otherwise it's hard to reason about it. These things should happen on corresponding pairs of events.

I see you are changing the policy when you call `uiWindow.makeKeyAndOrderFront`, so it seems the relevant lifecycle is becoming the key window. It's not scalable and effective to ensure the right policy in all places where the window can become key 

Instead, use the events. We should set regular when the window becomes key, and accessory when it's not key windows anymore. I see there are `didBecomeKeyNotification` and `didResignKeyNotification`.